lib/markdown-preview-view.js

```diff
@@ -0,0 +1,532 @@
+      this.subscribeToFilePath(this.filePath)
+    } else {
+      this.disposables.add(
+        atom.packages.onDidActivateInitialPackages(() => {
+          return this.subscribeToFilePath(this.filePath)
+        })
+      )
+    }
+  }
+
+  serialize () {
+    return {
+      deserializer: 'MarkdownPreviewView',
+      filePath: this.getPath() != null ? this.getPath() : this.filePath,
+      editorId: this.editorId
+    }
+  }
+
+  copy () {
+    return new MarkdownPreviewView({
+      editorId: this.editorId,
+      filePath: this.getPath() != null ? this.getPath() : this.filePath
+    })
+  }
+
+  destroy () {
+    this.disposables.dispose()
+    return this.element.remove()
+  }
+
+  registerScrollCommands () {
+    this.disposables.add(
+      atom.commands.add(this.element, {
+        'core:move-up': () => {
+          this.element.scrollTop -= document.body.offsetHeight / 20
+        },
+        'core:move-down': () => {
+          this.element.scrollTop += document.body.offsetHeight / 20
+        },
+        'core:page-up': () => {
+          this.element.scrollTop -= this.element.offsetHeight
+        },
+        'core:page-down': () => {
+          this.element.scrollTop += this.element.offsetHeight
+        },
+        'core:move-to-top': () => {
+          this.element.scrollTop = 0
+        },
+        'core:move-to-bottom': () => {
+          this.element.scrollTop = this.element.scrollHeight
+        }
+      })
+    )
+  }
+
+  onDidChangeTitle (callback) {
+    return this.emitter.on('did-change-title', callback)
+  }
+
+  onDidChangeModified (callback) {
+    // No op to suppress deprecation warning
+    return new Disposable()
+  }
+
+  onDidChangeMarkdown (callback) {
+    return this.emitter.on('did-change-markdown', callback)
+  }
+
+  subscribeToFilePath (filePath) {
+    this.file = new File(filePath)
+    this.emitter.emit('did-change-title')
+    this.disposables.add(
+      this.file.onDidRename(() => this.emitter.emit('did-change-title'))
+    )
+    this.handleEvents()
+    return this.renderMarkdown()
+  }
+
+  resolveEditor (editorId) {
+    const resolve = () => {
+      this.editor = this.editorForId(editorId)
+
+      if (this.editor != null) {
+        this.emitter.emit('did-change-title')
+        this.disposables.add(
+          this.editor.onDidDestroy(() =>
+            this.subscribeToFilePath(this.getPath())
+          )
+        )
+        this.handleEvents()
+        return this.renderMarkdown()
+      } else {
+        return this.subscribeToFilePath(this.filePath)
+      }
+    }
+
+    if (atom.packages.hasActivatedInitialPackages()) {
+      return resolve()
+    } else {
+      return this.disposables.add(
+        atom.packages.onDidActivateInitialPackages(resolve)
+      )
+    }
+  }
+
+  editorForId (editorId) {
+    for (const editor of atom.workspace.getTextEditors()) {
+      if (
+        (editor.id != null ? editor.id.toString() : undefined) ===
```

Comment:
Could this conditional be rewritten as `editor.id && editor.id.toString() === editorId.toString()`?

Reply:
Can `editor.id` be an empty `string`? If so I can rewrite it to `editor.id != null && editor.id.toString() === editorId.toString()`

Reply:
There shouldn't be a case where `editor.id` is null. It's supposed to be an ever-increasing integer according to https://github.com/atom/atom/blob/0cd702003662990a13cbb420a9141c58a43ec93f/src/text-editor.js#L130.